src/thing/thing.test.ts

```diff
@@ -801,6 +801,36 @@ describe("asIri", () => {
     );
   });
 
+  it("triggers a TypeScript error when passed a ThingLocal without a base IRI", () => {
+    const localThing: ThingLocal = createThing();
+
+    // @ts-expect-error
+    expect(() => asUrl(localThing)).toThrow();
+  });
+
+  // This currently fails because a plain Thing always has a `url` property that is a string,
+  // and is therefore indistinguishable from a ThingPersisted. Not sure what the solution is yet.
+  // Meanwhile TS users won't get a build-time error if they're passing a regular Thing,
+  // which is annoying but not a major issue.
```

Comment:
I'm a bit unclear on the terminology used here, i.e., 'plain thing', 'regular thing', 'persisted thing' - and it doesn't even mention a 'local thing'. So is a 'plain thing' the same as a 'regular thing' the same as a 'local thing'...?

Reply:
Yep, I reworded that to use "plain" twice instead of switching to "regular", and added backticks around the type names (`Thing` and `ThingPersisted`) to hopefully clarify I'm referring to those raw types: 2ba8d5ee.

Basically, we went from `ThingLocal` and `ThingPersisted` from being distinct types, and `Thing` being the union of those two`, to `ThingLocal` being a subset of `ThingPersisted` (i.e. it has a full URL, but that URL is a skolem that indicates that it's a local URL) and `Thing` being equal to `ThingPersisted`. Not great and I hope we'll find a solution to that at some point, but it's not a major issue either.